sdk/identity/azure-identity/azure/identity/_credentials/authorization_code.py

```diff
@@ -64,7 +64,15 @@ def get_token(self, *scopes, **kwargs):
             self._authorization_code = None  # auth codes are single-use
             return token
 
-        token = self._client.get_cached_access_token(scopes) or self._redeem_refresh_token(scopes, **kwargs)
+        token = self._client.get_cached_access_token(scopes)
+        if not token:
+            token = self._redeem_refresh_token(scopes, **kwargs)
+        elif self._client.should_refresh(token):
+            try:
+                self._redeem_refresh_token(scopes, **kwargs)
+            except Exception:  # pylint: disable=broad-except
+                pass
```

Comment:
Should we be logging refreshes which fail here? Is this already done in _redeem_refresh_token?

Reply:
This is a good question.

I am leaning towards not logging it because:
* if there is a valid token available, user will continue to use that one and there is no need to log it.
* if there is no valid token, user cannot get one and we will log that event (already implemented)